### DISCUSSION TITLE
Bypass sensors open when arm Ring

### DISCUSSION THREAD

**Discussion by alessandroarosio1977** [tsightler/ring-mqtt#522] [Support Q&A]
Hi, I'am trying to make a Nodered flow for arm at home my Ring alarm (integrated in HA with your addon) and it works fine normally.
But if there's a windows open, Rings do not arm and say to wait for bypass confirm.
There is a solution?

**Comment by jeffgraham-hub** [comment]
You can choose how it is handled unfer the settings of each device. On Jan 2, 2023 5:04 AM, alessandroarosio1977 ***@***.***> wrote:
Hi, I'am trying to make a Nodered flow for arm at home my Ring alarm (integrated in HA with your addon) and it works fine normally.
But if there's a windows open, Rings do not arm and say to wait for bypass confirm.
There is a solution?

**Comment by jeffgraham-hub** [comment]
Here is a screenshot of the setting. 

![Screenshot_20230102_081157_Home Assistant](https://user-images.githubusercontent.com/65316307/210246987-f68ca47d-805d-4fb1-adaf-4f44749413f3.jpg)

**Comment by tsightler** [comment]
Sensor bypass modes can be set on a per-sensor basis and the modes are documented here:

https://github.com/tsightler/ring-mqtt/wiki/Configuration-Details#sensor-bypass

Since you are using NodeRed you will need to find the topics yourself, but the overall format is documented in the wiki and I've added the specific paths for the bypass mode topics.

https://github.com/tsightler/ring-mqtt/wiki/MQTT-Device-Topics#contact-sensor

**Reply by mhirsch47** [reply]
I have a similar issue. Arm with bypass doesn't work any more. For testing, I opened one window and set the bypass mode to "Faulted". Arming with or without bypass option doesn't work. I'm using the "ordinary" HA automations.

Using current version of all components as of today.

```
"entity_id": "select.fenster_oben_bypass_mode",
            "icon": null,
            "original_device_class": null,
            "original_icon": null,
            "state": {
              "entity_id": "select.fenster_oben_bypass_mode",
              "state": "Faulted",
              "attributes": {
                "options": [
                  "Never",
                  "Faulted",
                  "Always"
                ],
                "friendly_name": "Fenster Oben Bypass Mode"
```

**Reply by tsightler** [reply]
What do the logs say?

**Reply by tsightler** [reply]
Also, what do you mean by arming with or without bypass option?  What bypass option are you referring to?

I just tested this with one of my contact sensors and it worked fine so I am not able to reproduce, nor have there been any other reports of issues with this feature (even this thread does not report a problem, other than the user not knowing how to use the feature).

The logs should be really clear about what is happening, for example, here's the output of my attempt to arm my alarm in away mode with one door opened:

```
2023-08-17T21:30:28.982Z ring-mqtt [Sightler Acres Alarm] Received set alarm mode ARM_AWAY for location Sightler Acres (a9ce28cb-fafb-421c-4508-29d54f48cbd3)
2023-08-17T21:30:28.982Z ring-mqtt [Sightler Acres Alarm] The following sensors will be bypassed [Reason]: Sliding Glass Door [Faulted]
2023-08-17T21:30:28.982Z ring-mqtt [Sightler Acres Alarm] Set alarm mode: ARM_AWAY
```

You can see that it correctly detected the sensor as faulted and, since bypass mode for that sensor was set to "Faulted", it automatically bypassed that sensor during arming.

Also, you state that it "doesn't work any more", which implies that it did work at some point?  When was the last time you know that it was working?  Pretty much nothing has changed in this code since the 5.x release which was 18 months ago.

**Reply by mhirsch47** [reply]
Works for me as well now. 2 mistakes made:
- I thought I need to use the "ARM_BYPASS" service
- I thought the setting for bypass:Faulted is a permanent setting.
I have the Faulted setting now in my automation just before calling ARM_AWAY and it works as expected.
thx!

**Reply by tsightler** [reply]
Setting faulted should be persistent as the setting is stored in the ring-state.json file.  If it is not, then that would be a bug, but I'm not able to reproduce this.  How do you have ring-mqtt deployed?
